components/script/dom/servoparser/async_html.rs

```diff
@@ -0,0 +1,561 @@
+                            elem.set_attribute_from_parser(attr.name, DOMString::from(String::from(attr.value)), None);
+                        }
+                    },
+                    None => panic!("Node not found!")
+                }
+            }
+            ParseOperation::RemoveFromParent(target) => {
+                match self.nodes.borrow().get(&target) {
+                    Some(target) => {
+                        if let Some(ref parent) = target.GetParentNode() {
+                            parent.RemoveChild(&*target).unwrap();
+                        }
+                    }
+                    None => panic!("Node not found!")
+                }
+            }
+            ParseOperation::MarkScriptAlreadyStarted(node) => {
+                match self.nodes.borrow().get(&node) {
+                    Some(node) => {
+                        let script = node.downcast::<HTMLScriptElement>();
+                        script.map(|script| script.set_already_started(true));
+                    }
+                    None => panic!("Node not found!")
+                }
+            }
+            ParseOperation::ReparentChildren(parent, new_parent) => {
+                let nodes = self.nodes.borrow();
+                let parent = nodes.get(&parent).unwrap();
+                let new_parent = nodes.get(&new_parent).unwrap();
+                while let Some(ref child) = parent.GetFirstChild() {
+                    new_parent.AppendChild(&child).unwrap();
+                }
+            }
+            ParseOperation::AssociateWithForm(target, form) => {
+                let nodes = self.nodes.borrow();
+                let node = nodes.get(&target).unwrap();
+                let form = nodes.get(&form).unwrap();
+
+                let form = Root::downcast::<HTMLFormElement>(Root::from_ref(&**form))
+                    .expect("Owner must be a form element");
+
+                let elem = node.downcast::<Element>();
+                let control = elem.and_then(|e| e.as_maybe_form_control());
+
+                if let Some(control) = control {
+                    control.set_form_owner_from_parser(&form);
+                } else {
+                    // TODO remove this code when keygen is implemented.
+                    assert!(node.NodeName() == "KEYGEN", "Unknown form-associatable element");
+                }
+            }
+            ParseOperation::Pop(node) => {
+                let nodes = self.nodes.borrow();
+                let node = nodes.get(&node).unwrap();
+                let node = Root::from_ref(&**node);
+                vtable_for(&node).pop();
+            }
+            ParseOperation::CreatePI(node) => {
+                let parse_node_data = self.parse_node_data.borrow();
+                let data = parse_node_data.get(&node).unwrap();
+                let pi = ProcessingInstruction::new(
+                    DOMString::from(data.target.clone().unwrap()),
```

Comment:
Nit: instead of cloning and then wrapping, you could use `DOMString::from` with a `&str` argument.